ceno_zkvm/src/tables/ram/ram_circuit.rs

```diff
@@ -128,17 +128,20 @@ pub trait DynVolatileRamTable {
     const RAM_TYPE: RAMType;
     const V_LIMBS: usize;
 
-    const OFFSET_ADDR: Addr;
-    const END_ADDR: Addr;
+    // const OFFSET_ADDR: Addr;
+    // const END_ADDR: Addr;
+
+    fn offset_addr(platform: &Platform) -> Addr;
+    fn end_addr(platform: &Platform) -> Addr;
```

Comment:
`end_addr` is not actually needed in the trait.

Reply:
It's used in `max_len()` (the overwritten impl as well), no?